### ISSUE TITLE
Multipolygon relations: not all borders are labeled

### ISSUE THREAD

**Issue by gpsvisualizer** [gravitystorm/openstreetmap-carto#3152]
If a multipolygon relation includes multiple discrete parcels of land, the name of the relation is ONLY drawn around the borders of the largest parcel.  This is confusing in situations where some of the land might be far out of view of the biggest chunk.

I supposed I can understand why only one "area label" is drawn per relation; it might look weird if there were fifty parcels visible, all with the same label... but I think the borders, which only appear when zoomed in, should be labeled on every parcel.

![multipolygon rendering annotated](https://user-images.githubusercontent.com/8356259/38064301-2395b6bc-32b2-11e8-877d-69be6dfe3d98.png)

Attached file comes from here: https://www.openstreetmap.org/#map=15/45.7934/-121.6350

**Comment by matkoniecz** [comment]
From what I see it is request to revert #59

**Comment by gpsvisualizer** [comment]
Not exactly. #59 is about whether or not to label each AREA that is part of a multipolygon; I agree that only one area label is best.

Here, I'm asking for the BORDERS to be labeled on all of the areas.  This is something that only happens at high zoom levels, and there's no reason it should be restricted to one area per relation.

**Comment by chazanov** [comment]
So @gpsvisualizer you wan't to make the rendering size-dependent?

**Comment by gpsvisualizer** [comment]
Again, I'm not talking about the area labels, I'm talking about the small text that flows along the borders -- text which is ONLY visible at high zoom levels.  Even if a forest has dozens or separate areas, all of those areas should have their borders labeled when zoomed in.

**Comment by chazanov** [comment]
It begins to show up at z=15. So let's just try it!

**Comment by gpsvisualizer** [comment]
This is still a problem.  Any hope of a solution?

**Comment by jeisenbe** [comment]
It looks like this same problem happens with administrative boundaries too, at least in my test environment.

Are there any other types of multipolygons that have text labels on their borders, besides administrative, nature_reserve, national_park and protected_area?

Before
![province-multipolygon-test-before](https://user-images.githubusercontent.com/42757252/51380853-0403b500-1b56-11e9-99ce-1e5d6d492c70.png)

After
![province-multipolygon-test-after](https://user-images.githubusercontent.com/42757252/51380962-3b726180-1b56-11e9-9d65-24c719988238.png)

**Comment by gpsvisualizer** [comment]
Does your quick-'n'-easy solution cause an AREA label to appear in all of the sub-polygons too?  Because I think that issue was more up for debate than just the edge labels... I think it'd be best to stick with labeling just the edges for now.

**Comment by jeisenbe** [comment]
> Does your quick-'n'-easy solution cause an AREA label to appear in all of
the sub-polygons too?

No, this only affects the admin-text and nature-reserve-text layers

**Comment by gpsvisualizer** [comment]
So can you make it happen for real?

**Comment by gpsvisualizer** [comment]
Hallelujah. Looks like you solved multiple problems. I look forward to it going live.
